src/core/basic_client.rs

```diff
@@ -672,6 +673,52 @@ impl BasicClient {
         }
     }
 
+    /// **[Cryptographic Operation]** Perform a raw key agreement.
+    ///
+    /// The provided private key **must** have its `derive` flag set
+    /// to `true` in its [key policy](https://docs.rs/parsec-interface/*/parsec_interface/operations/psa_key_attributes/struct.Policy.html).
+    ///
+    /// The raw_key_agreement will be performed with the algorithm defined in `alg`,
+    /// but only after checking that the key policy and type conform with it.
+    ///
+    /// `peer_key` must be the peer public key to use in the raw key derivation. It must
+    /// be in a format supported by [`PsaImportKey`](https://parallaxsecond.github.io/parsec-book/parsec_client/operations/psa_import_key.html).
```

Comment:
I'd like to see us supporting Rust native types for this so we don't have to accept raw slices of bytes.... (not in this PR, obviously)

Reply:
Agreed, giving byte buffer makes me uncomfortable 😅